inc/ChangeLog/ChangeLog.php

```diff
@@ -804,4 +772,159 @@ public function getExternalEditRevInfo()
+
+        if (!$fileRev && $lastRev) {                // item file does not exist
+            // check consistency against changelog
+            $revInfo = $this->getRevisionInfo($lastRev);
+            if ($revInfo['type'] == DOKU_CHANGE_TYPE_DELETE) {
+                $this->currentRevision = $lastRev;
+                $this->cache[$this->id][$this->currentRevision] += [
+                    'current' => true,
+                ];
+                return $this->getRevisionInfo($this->currentRevision);
+            }
+
+            // externally deleted
+            $revInfo = [
+                'date' => $lastRev +1,
+                'ip'   => '127.0.0.1',
+                'type' => DOKU_CHANGE_TYPE_DELETE,
+                'id'   => $this->id,
+                'user' => '',
+                'sum'  => $lang['deleted'] .' - '. $lang['external_edit'],
+                'extra' => '',
+                'sizechange' => -io_getSizeFile($this->getFilename($lastRev)),
+                'current' => true,
+                'timestamp' => 'unknown',
+            ];
+
+        } elseif ($fileRev) {                       // item file exist
+            // here, file timestamp is different with last revision in changelog
+            $isJustCreated = !boolval($lastRev);
+            $filesize_new = filesize($this->getFilename());
+            $filesize_old = $lastRev ? io_getSizeFile($this->getFilename($lastRev)) : 0;
+            $sizechange = $filesize_new - $filesize_old;
+
+            if ($isJustCreated) { // lastRev is null
+                $rev = $timestamp = $fileRev;
+                $sum = $lang['created'] .' - '. $lang['external_edit'];
+            } elseif ($fileRev > $lastRev) {
+                $rev = $timestamp = $fileRev;
+                $sum = $lang['external_edit'];
+            } else {
+                // $fileRev is older than $lastRev, externally reverted an old file
+                $rev = max($fileRev, $lastRev +1);
+                $timestamp = 'unknown';
+                $sum = $lang['external_edit'];
+            }
+
+            // externally created or edited
+            $revInfo = [
+                'date' => $rev,
+                'ip'   => '127.0.0.1',
+                'type' => $isJustCreated ? DOKU_CHANGE_TYPE_CREATE : DOKU_CHANGE_TYPE_EDIT,
+                'id'   => $this->id,
+                'user' => '',
+                'sum'  => $sum,
+                'extra' => '',
+                'sizechange' => $sizechange,
+                'current' => true,
+                'timestamp' => $timestamp,
+            ];
+        }
+
+        // cache current revision information of external edition
+        $this->currentRevision = $revInfo['date'];
+        $this->cache[$this->id][$this->currentRevision] = $revInfo;
+        return $this->getRevisionInfo($this->currentRevision);
+    }
+
+    /**
+     * Checks if the revision is external edition
+     * @return boolean
+     */
+    public function isExternalEdition($rev =null)
```

Comment:
I thought there are two types of "**external revision**", 1) which is pending to register into changelog to avoid multiple records by repeated external edits and view cycle, and 2) already registered in changelog. Here I had referred former as **external edition**. Anyway, I will use `isCurrentRevision()` and `isLastRevision()` instead of `isExternalRevision()`.